programs/marginfi/src/instructions/marginfi_group/collect_bank_fees.rs

```diff
@@ -399,3 +399,131 @@ pub struct LendingPoolWithdrawInsurance<'info> {
+
+    pub admin: Signer<'info>,
+
+    /// Bank fees will be sent to the canonical ATA of this wallet.
+    ///
+    /// CHECK: Completely unchecked, admin picks a destination without restrictions
+    pub destination_account: AccountInfo<'info>,
+}
+
+pub fn lending_pool_withdraw_fees_permissionless<'info>(
+    mut ctx: Context<'_, '_, 'info, 'info, LendingPoolWithdrawFeesPermissionless<'info>>,
+    amount: u64,
+) -> MarginfiResult {
+    let LendingPoolWithdrawFeesPermissionless {
+        bank: bank_loader,
+        fee_vault,
+        fee_vault_authority,
+        dst_token_account,
+        token_program,
+        ..
+    } = ctx.accounts;
+
+    let bank = bank_loader.load()?;
+
+    let fees_dest_wallet = &bank.fees_destination_account;
+    let fees_mint = &bank.mint;
+    let fees_token_program = &token_program.key();
+
+    // Ensure that the fees_destination_account was previously set by the group admin
+    check!(
+        !fees_dest_wallet.eq(&Pubkey::default()),
+        MarginfiError::InvalidFeesDestinationAccount
+    );
+
+    // Ensure the destination is the canonical ATA of the user-specified wallet
+    let ata_expected = get_associated_token_address_with_program_id(
+        fees_dest_wallet,
+        fees_mint,
+        fees_token_program,
+    );
+    check_eq!(
+        ata_expected,
+        dst_token_account.key(),
+        MarginfiError::InvalidFeesDestinationAccount
+    );
+
+    let maybe_bank_mint =
+        utils::maybe_take_bank_mint(&mut ctx.remaining_accounts, &bank, fees_token_program)?;
+
+    bank.withdraw_spl_transfer(
+        amount,
+        fee_vault.to_account_info(),
+        dst_token_account.to_account_info(),
+        fee_vault_authority.to_account_info(),
+        maybe_bank_mint.as_ref(),
+        token_program.to_account_info(),
+        bank_signer!(
+            BankVaultType::Fee,
+            bank_loader.key(),
+            bank.fee_vault_authority_bump
+        ),
+        ctx.remaining_accounts,
+    )?;
+
+    Ok(())
+}
+
+#[derive(Accounts)]
+pub struct LendingPoolWithdrawFeesPermissionless<'info> {
+    pub group: AccountLoader<'info, MarginfiGroup>,
+
+    #[account(
+        has_one = group
+    )]
+    pub bank: AccountLoader<'info, Bank>,
+
+    /// CHECK: ⋐ ͡⋄ ω ͡⋄ ⋑
+    #[account(
+        mut,
+        seeds = [
+            FEE_VAULT_SEED.as_bytes(),
+            bank.key().as_ref(),
+        ],
+        bump = bank.load()?.fee_vault_bump
+    )]
+    pub fee_vault: AccountInfo<'info>,
+
+    /// CHECK: ⋐ ͡⋄ ω ͡⋄ ⋑
+    #[account(
+        seeds = [
+            FEE_VAULT_AUTHORITY_SEED.as_bytes(),
+            bank.key().as_ref(),
+        ],
+        bump = bank.load()?.fee_vault_authority_bump
+    )]
+    pub fee_vault_authority: AccountInfo<'info>,
+
+    /// CHECK: Handler will validate this is a canonical ATA of the `fees_destination_account`
+    /// registered on `bank`
+    #[account(mut)]
+    pub dst_token_account: AccountInfo<'info>,
```

Comment:
Add a has_one check for this on the bank if we decide to store the ATA instead of the wallet, this simplifies the checks you need to in the handler. You can also add an address constraint that validates it's not pubkey default.